app/scripts/controllers/permissions/index.js

```diff
@@ -510,6 +507,100 @@ export class PermissionsController {
     })
   }
 
+  /**
+   * Removes all known domains and their related permissions.
+   */
+  clearPermissions () {
+    this.permissions.clearDomains()
+    this.notifyAllDomains({
+      method: NOTIFICATION_NAMES.accountsChanged,
+      result: [],
+    })
+  }
+
+  /**
+   * Stores domain metadata for the given origin (domain).
+   * Deletes metadata for domains without permissions in a FIFO manner.
+   * Metadata is only deleted for domains with permissions on boot in order to
+   * prevent a degraded user experience, as metadata currently can't be
+   * requested on-demand.
```

Comment:
Nit: I'm having trouble understanding this description, and I'm not sure it's all true anymore?

Something like this would make more sense to me:

```suggestion
   * If the pending site metadata exceeds the max cache size, the oldest metadata
   * will be deletedto make room.
```

This makes it more clear _when_ and _why_ metadata is deleted, and omits the detail about what happens on boot, which doesn't seem relevant to this function.

Reply:
I updated this comment in d3c7d89, please let me know if it suffices.

Reply:
Seems fine to me!